crates/load-test/src/main.rs

```diff
@@ -0,0 +1,450 @@
+                "calldata": call_data,
+                "entry_point_selector": entry_point_selector,
+            },
+            "block_hash": at_block,
+        }),
+    )
+    .await
+}
+
+async fn post_jsonrpc_request<T: DeserializeOwned>(
+    user: &mut GooseUser,
+    method: &str,
+    params: serde_json::Value,
+) -> MethodResult<T> {
+    let request = jsonrpc_request(method, params);
+    let response = user.post_json("", &request).await?.response?;
+    #[derive(Deserialize)]
+    struct TransactionReceiptResponse<T> {
+        result: T,
+    }
+    let response: TransactionReceiptResponse<T> = response.json().await?;
+
+    Ok(response.result)
+}
+
+fn jsonrpc_request(method: &str, params: serde_json::Value) -> serde_json::Value {
+    json!({
+        "jsonrpc": "2.0",
+        "id": "0",
+        "method": method,
+        "params": params,
+    })
+}
+
+#[tokio::main]
+async fn main() -> Result<(), GooseError> {
+    GooseAttack::initialize()?
+        // primitive operations using the database
+        .register_scenario(
+            scenario!("block_by_number").register_transaction(transaction!(task_block_by_number)),
+        )
+        .register_scenario(
+            scenario!("block_by_hash").register_transaction(transaction!(task_block_by_hash)),
+        )
+        .register_scenario(
+            scenario!("block_transaction_count_by_hash")
+                .register_transaction(transaction!(task_block_transaction_count_by_hash)),
+        )
+        .register_scenario(
+            scenario!("block_transaction_count_by_number")
+                .register_transaction(transaction!(task_block_transaction_count_by_number)),
+        )
+        .register_scenario(
+            scenario!("transaction_by_hash")
+                .register_transaction(transaction!(task_transaction_by_hash)),
+        )
+        .register_scenario(
+            scenario!("transaction_by_block_number_and_index")
+                .register_transaction(transaction!(task_transaction_by_block_number_and_index)),
+        )
+        .register_scenario(
+            scenario!("transaction_by_block_hash_and_index")
+                .register_transaction(transaction!(task_transaction_by_block_hash_and_index)),
+        )
+        .register_scenario(
+            scenario!("transaction_receipt_by_hash")
+                .register_transaction(transaction!(task_transaction_receipt_by_hash)),
+        )
+        .register_scenario(
+            scenario!("block_number").register_transaction(transaction!(task_block_number)),
+        )
+        .register_scenario(
+            scenario!("get_events").register_transaction(transaction!(task_get_events)),
+        )
+        // primitive operations that don't use the database
+        .register_scenario(scenario!("syncing").register_transaction(transaction!(task_syncing)))
+        .register_scenario(scenario!("call").register_transaction(transaction!(task_call)))
```

Comment:
This doesn't read from the database within pathfinder, but it does from the python side (?).

Reply:
Fixed, thanks!